dracut/modules.d/99kiwi-lib/kiwi-lib.sh

```diff
@@ -42,22 +41,31 @@ function lookup_disk_device_from_root {
     fi
     for disk_device in $(
         lsblk -p -n -r -s -o NAME,TYPE "${root_device}" |\
-            grep disk | cut -f1 -d ' '
+            grep -E "disk|raid" | cut -f1 -d ' '
     ); do
-        disk_matches=$((disk_matches + 1))
-    done
-    # Check if root_device is managed by multipath. If this
-    # is the case prefer the multipath mapped device because
-    # directly accessing the mapped devices is no longer
-    # possible
-    if type multipath &> /dev/null; then
-        for wwn in $(multipath -l -v1 "${disk_device}");do
-            if [ -e "/dev/mapper/${wwn}" ];then
-                disk_device="/dev/mapper/${wwn}"
-                break
+        # Check if root_device is managed by multipath. If this
+        # is the case prefer the multipath mapped device because
+        # directly accessing the mapped devices is no longer
+        # possible
+        if type multipath &> /dev/null; then
+            for wwn in $(multipath -l -v1 "${disk_device}");do
+                if [ -e "/dev/mapper/${wwn}" ];then
+                    disk_device="/dev/mapper/${wwn}"
+                    echo "${disk_device}"
+                    return
+                fi
+            done
+        fi
+        # Check if root_device is managed by mdadm. If this
+        # is the case prefer the md device over the storage
+        # disks because this would create conflicts otherwise
+        if type mdadm &> /dev/null; then
+            if mdadm --detail -Y "${disk_device}" 2>/dev/null; then
```

Comment:
This is good. Maybe a 
```
>/dev/null 2>&1
```
prevents the output from spoiling with unneeded information about the md-device (Additionally, the warning message, that any device isn't manageable by mdadm is stdout).

Reply:
yep I changed it to &>/dev/null